### ISSUE TITLE
[Feature]: KITTI D1 disparity error metric (follow-up to #1123)

### ISSUE THREAD

**Issue by VanniLeonardo** [kornia/kornia#3758]
### 🚀 Feature Description

Add `kitti_d1_error` to `kornia.metrics`, completing the stereo disparity metrics started in #1123. D1 is the KITTI benchmark's primary stereo metric: the fraction of pixels whose disparity error exceeds **both** an absolute threshold (3 px) **and** a relative threshold (5%). It builds directly on the `mean_bad_pixel_error` / shared validation+reduction helpers already merged.

### 📂 Feature Category

Geometry

### 💡 Motivation

#1123 added MAE, RMSE, and bad-pixel error but intentionally deferred KITTI D1 to keep that PR focused. D1 (a.k.a. D1-all) is the headline ranking metric on the KITTI 2015 stereo benchmark, so stereo researchers comparing against published numbers need it. Without it they re-implement it per-project.

### 💭 Proposed Solution

New functional metric in `kornia/metrics/disparity.py`, reusing the existing `_check_disparity_inputs` / `_reduce_disparity_error` helpers and matching the signatures of the merged metrics:

```python
def kitti_d1_error(
    input: torch.Tensor,
    target: torch.Tensor,
    abs_threshold: float = 3.0,
    rel_threshold: float = 0.05,
    valid_mask: torch.Tensor | None = None,
    reduction: str = "mean",
) -> torch.Tensor:
    # bad if |d - d_gt| > abs_threshold AND |d - d_gt| / |d_gt| > rel_threshold
```
Exported from `kornia/metrics/__init__.py`, documented in `docs/source/metrics.rst` (Stereo section), and tested in `tests/metrics/test_disparity.py` with hardcoded expected values.

**Secondary (optional,, would like to ask a maintainer):** `nn.Module` wrappers for the four disparity metrics, mirroring the existing `AEPE`/`aepe` precedent. Happy to include or split into a separate PR.

Per recent reviewer feedback, the new function(s) will be verified `torch.compile`-compatible (dynamo test).

### 🔄 Alternatives Considered

Leaving D1 to downstream users (status quo, causes re-implementation); or folding a `relative` flag into `mean_bad_pixel_error` (rejected: D1's dual abs-AND-relative criterion is a distinct, named metric and clearer as its own function).

### 🎯 Use Cases

Reporting/validating stereo-matching models against KITTI 2015 leaderboard numbers; training-loop monitoring of disparity quality.

## 📝 Additional Context

Reference: KITTI 2015 stereo benchmark, Menze & Geiger, *Object Scene Flow for Autonomous Vehicles*, CVPR 2015 - https://www.cvlibs.net/datasets/kitti/eval_scene_flow.php?benchmark=stereo. Follow-up to #1123.

### 🤝 Contribution Intent

- [x] I plan to submit a PR to implement this feature
- [ ] I'm requesting this feature but not planning to implement it

**Comment by Shanmuk4622** [comment]
Hi @VanniLeonardo — I’d be happy to take this on if it’s still open. I’m new to contributing here and excited to get involved!

I’ve reviewed the existing `kornia/metrics/disparity.py` patterns (`_check_disparity_inputs`, `_reduce_disparity_error`, and `mean_bad_pixel_error`) along with the KITTI D1 definition from Menze & Geiger (CVPR 2015). The dual AND‑criterion (‖d − d_gt‖ > 3.0 **and** ‖d − d_gt‖ / ‖d_gt‖ > 0.05) fits cleanly on top of what’s already implemented.

**Plan for the PR**
- Add `kitti_d1_error` to `kornia/metrics/disparity.py`, reusing `_check_disparity_inputs` and `_reduce_disparity_error`, following the same signature style as `mean_bad_pixel_error`
- Export from `kornia/metrics/__init__.py`
- Include a docstring with the mathematical definition and a runnable example
- Add tests in `tests/metrics/test_disparity.py` with hardcoded expected values (covering edge cases: all pixels correct, all pixels incorrect, `valid_mask` usage, and a numerically verified mixed case)
- Update `docs/source/metrics.rst` under the Stereo section
- Verify `torch.compile` (dynamo) compatibility as noted in the issue

On the `nn.Module` wrapper: I can also add a `KITTiD1Error` module in the same PR, mirroring the AEPE precedent, but I’ll wait for maintainer guidance before including it.

Would a maintainer be able to assign this? I’d be glad to start right away.

**Comment by VanniLeonardo** [comment]
Hi @Shanmuk4622 , I opened this as a follow-up to my own PR (#1123) and flagged it as something I'm implementing myself, so I'll be taking this one. I already have the helpers (`_check_disparity_inputs`, `_reduce_disparity_error`) in place from the previous PR, so I'm set to start once a maintainer assigns me.

**Comment by ducha-aiki** [comment]
@VanniLeonardo Please, go ahead
